packages/@webex/plugin-meetings/src/controls-options-manager/index.ts

```diff
@@ -135,30 +135,43 @@ export default class ControlsOptionsManager {
 
   /**
    * @param {Setting} setting
-   * @param {boolean} enabled
    * @private
    * @memberof ControlsOptionsManager
    * @returns {Promise}
    */
-  private setControls(setting: Setting, enabled: boolean): Promise<any> {
-    LoggerProxy.logger.log(`ControlsOptionsManager:index#setControls --> ${setting} [${enabled}]`);
-
-    if (Util?.[`${enabled ? CAN_SET : CAN_UNSET}${setting}`](this.displayHints)) {
-      // @ts-ignore
-      return this.request.request({
-        uri: `${this.locusUrl}/${CONTROLS}`,
-        body: {
-          [camelCase(setting)]: {
-            [ENABLED]: enabled,
-          },
-        },
-        method: HTTP_VERBS.PATCH,
-      });
+  private setControls(setting: {[key in Setting]?: boolean}): Promise<any> {
+    const muteUnmuteAll = Object.keys(setting).includes(Setting.muted);
+
+    const body = muteUnmuteAll ? {audio: {}} : {};
+    for (const [key, value] of Object.entries(setting)) {
+      LoggerProxy.logger.log(`ControlsOptionsManager:index#setControls --> ${key} [${value}]`);
+      if (muteUnmuteAll) {
```

Comment:
IMO this should be part of a list of checks (put it in an array of strings) and check to see if array of allowlist types (or an array of objects) - as that would be more dynamic and resistant to changes